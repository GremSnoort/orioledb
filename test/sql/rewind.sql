CREATE SCHEMA rewind;
SET SESSION search_path = 'rewind';
CREATE EXTENSION orioledb;
\set VERBOSITY terse
\set VERBOSITY default

CREATE table heap_table (i int) USING heap;
CREATE TABLE o_test_rewind (i int) USING orioledb;

INSERT INTO heap_table VALUES (1);

INSERT INTO o_test_rewind VALUES (1);
INSERT INTO o_test_rewind VALUES (2);
INSERT INTO o_test_rewind VALUES (3);
INSERT INTO o_test_rewind VALUES (4);
INSERT INTO o_test_rewind VALUES (5);

-- Should fail
select orioledb_rewind_by_time(600);
select orioledb_rewind_by_time(0);

BEGIN;
INSERT INTO heap_table VALUES (2);
PREPARE TRANSACTION 'prep';
ROLLBACK;

select orioledb_rewind_by_time(10);

COMMIT PREPARED 'prep';

DROP TABLE heap_table;
DROP TABLE o_test_rewind;

DROP EXTENSION orioledb CASCADE;
DROP SCHEMA rewind CASCADE;
RESET search_path;
