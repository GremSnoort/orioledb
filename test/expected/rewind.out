CREATE SCHEMA rewind;
SET SESSION search_path = 'rewind';
CREATE EXTENSION orioledb;
\set VERBOSITY terse
\set VERBOSITY default
CREATE table heap_table (i int) USING heap;
CREATE TABLE o_test_rewind (i int) USING orioledb;
INSERT INTO heap_table VALUES (1);
INSERT INTO o_test_rewind VALUES (1);
INSERT INTO o_test_rewind VALUES (2);
INSERT INTO o_test_rewind VALUES (3);
INSERT INTO o_test_rewind VALUES (4);
INSERT INTO o_test_rewind VALUES (5);
-- Should fail
select orioledb_rewind_by_time(600);
ERROR:  Requested rewind 600 s exceeds rewind_max_time 500 s
DETAIL:  request rewind less than 500 s back
select orioledb_rewind_by_time(0);
WARNING:  Zero or negative rewind requested, do nothing
 orioledb_rewind_by_time 
-------------------------
 
(1 row)

BEGIN;
INSERT INTO heap_table VALUES (2);
PREPARE TRANSACTION 'prep';
ROLLBACK;
WARNING:  there is no transaction in progress
select pg_sleep(2);
 pg_sleep 
----------
 
(1 row)

select orioledb_rewind_by_time(1);
ERROR:  Can't rewind due to prepared transactions in the cluster
COMMIT PREPARED 'prep';
DROP TABLE heap_table;
DROP TABLE o_test_rewind;
DROP EXTENSION orioledb CASCADE;
DROP SCHEMA rewind CASCADE;
RESET search_path;
